### PR TITLE
Press gallery fixes

### PR DESCRIPTION
**Pull request by jdtoombs**
- group date content on frontend instead of sending 7 days worth of elastic requests
- allow user to enter each filter, not one or the other
- clean up some code

Leaving in draft as I am playing with some things still. 

Also correct me if I am wrong - but grouping based on contributor aliases in headline, byline, body etc.. on the frontend would probably be slower than utilizing elastic search? 

Date is easy enough to group; however, I figured it may be best to get the hits from individual elastic search queries still (for each press member). Going to try and get an aggregate response for just the hits on that query. (in progress)

Another option which I would be a fan of.... is removing the hits completely from the drop down lists. Especially since the user can apply both filters at once now it just adds unnecessary complexity imo.

### CODE REVIEW

File: app/subscriber/src/features/press-gallery/PressGallery.tsx [hunk 1]
```diff
@@ -92,6 +88,23 @@ export const PressGallery: React.FC = () => {
     setDateOptions(dates);
   }, []);
 
+  // group content by date on the frontend instead of addtioanl fetches for each date
+  React.useEffect(() => {
+    if (!!content.length) {
+      const grouped = content.reduce((acc: any, content) => {
```
Comment (jdtoombs) [inline app/subscriber/src/features/press-gallery/PressGallery.tsx:94]: fix

File: app/subscriber/src/features/press-gallery/PressGallery.tsx [hunk 2]
```diff
@@ -34,10 +34,12 @@ export const PressGallery: React.FC = () => {
 
   const [content, setContent] = React.useState<IContentSearchResult[]>([]);
   const [pressMembers, setPressMembers] = React.useState<IPressMember[]>([]);
+  const [initialLoad, setInitialLoad] = React.useState(false);
   const [selected, setSelected] = React.useState<IContentModel[]>([]);
   const [dateOptions, setDateOptions] = React.useState<IDateOptions[]>([]);
   const [aliases, setAliases] = React.useState<string[]>([]);
   const [loading, setLoading] = React.useState(false);
+  const [groupedContent, setGroupedContent] = React.useState<any[]>([]);
```
Comment (jdtoombs) [inline app/subscriber/src/features/press-gallery/PressGallery.tsx:42]: fix

Comment (Fosol) [reply app/subscriber/src/features/press-gallery/PressGallery.tsx:42]: Don't use `any`

**Comment by jdtoombs** [comment]
have an idea closing for now

File: app/subscriber/src/features/press-gallery/PressGallery.tsx (hunk 2 above)
Comment (jdtoombs) [reply app/subscriber/src/features/press-gallery/PressGallery.tsx:42]: thats what the fix is for :stuck_out_tongue:, closing this PR for now though have an idea

Comment (jdtoombs) [reply app/subscriber/src/features/press-gallery/PressGallery.tsx:42]: (was telling myself to fix)

**Pull request closed without merging**